idaes/core/util/model_diagnostics.py

```diff
@@ -3559,7 +3559,7 @@ class IpoptConvergenceAnalysis:
 
     CONFIG = CACONFIG()
 
-    def __init__(self, model, **kwargs):
+    def __init__(self, model, solver=None, **kwargs):
```

Comment:
Could you please also change the name from `solver` to `solver_obj`? Typically if we had `solver` as the default argument, we'd pass the string `"ipopt"` instead of an IPOPT solver object.